core/src/main/java/org/openstack4j/openstack/storage/block/domain/AbstractCinderVolumeTypeEncryption.java

```diff
@@ -0,0 +1,164 @@
+        return provider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getVolumeTypeId() {
+        return volumeTypeId;
+    }
+
+    /**
+    * {@inheritDoc}
+    */
+    @Override
+    public String getCipher() {
+        return cipher;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getKeySize() {
+        return (keySize == null) ? 0 : keySize;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getControlLocation() {
+        return controlLocation;
+    }
+
+    @Override
+    public VolumeTypeEncryptionBuilder toBuilder() {
+        return new VolumeTypeEncryptionConcreteBuilder(this);
+    }
+
+    /**
+     * @return the Encryption Type Builder
+     */
+    public static VolumeTypeEncryptionBuilder builder() {
+        return new VolumeTypeEncryptionConcreteBuilder();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this).omitNullValues().add("id", id).add("volumeTypeId", volumeTypeId)
+                .add("provider", provider).add("cipher", cipher).add("keySize", keySize)
+                .add("controlLocation", controlLocation).toString();
+    }
+
+    public static class VolumeTypeEncryptionConcreteBuilder implements VolumeTypeEncryptionBuilder {
+
+        private CinderVolumeTypeEncryption cinderVolumeTypeEncryption;
+
+        VolumeTypeEncryptionConcreteBuilder() {
+            this(new CinderVolumeTypeEncryption());
+        }
+
+        VolumeTypeEncryptionConcreteBuilder(VolumeTypeEncryption volumeTypeEncryption) {
+            this.cinderVolumeTypeEncryption = (CinderVolumeTypeEncryption) volumeTypeEncryption;
+        }
+
+        @Override
+        public VolumeTypeEncryption build() {
+            return cinderVolumeTypeEncryption;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder from(VolumeTypeEncryption in) {
+            cinderVolumeTypeEncryption = (CinderVolumeTypeEncryption) in;
+            return this;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder provider(String provider) {
+            cinderVolumeTypeEncryption.provider = provider;
+            return this;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder id(String id) {
+            cinderVolumeTypeEncryption.id = id;
+            return this;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder volumeTypeId(String volTypeId) {
+            cinderVolumeTypeEncryption.volumeTypeId = volTypeId;
+            return this;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder cipher(String cipher) {
+            cinderVolumeTypeEncryption.cipher = cipher;
+            return this;
+        }
+
+        @Override
+        public VolumeTypeEncryptionBuilder keySize(int keySize) {
+            cinderVolumeTypeEncryption.keySize = keySize;
```

Comment:
I do not quite like the way the `None` is treated as `0` for key size, but you have made a good point it is done this way elsewhere as well. However, how does one create and encryption type with `None` key size (is that even supported use-case)? I suspect 0 will be sent.

We either have to to the reverse translation (`0` -> `null`) or not to (ab)use `0` to mean none.

Reply:
It is a supported usecase to create encryption without key size. Please see below:
"cinder encryption-type-create dc02a80d-1c7b-48e6-a55f-97305b64eed8 nov.volume.x.encryptor_provider"
This command worked and below is the response when I fetched the created encryption:
RESP BODY: {"volume_type_id": "dc02a80d-1c7b-48e6-a55f-97305b64eed8", "control_location": "front-end", "deleted": false, "created_at": "2020-03-31 09:27:08.000000", "updated_at": null, "encryption_id": "c4fb0888-98aa-4688-a9d8-1c2a30057888", "key_size": null, "provider": Encryptor", "deleted_at": null, "cipher": null}

It shows key-size as null.
Please suggest how to proceed.

Reply:
@jsaroja66, I believe we are talking about slightly different things. What you attached actually proves we can fetch and read the entity its key size is not configured. Than is great.

What I had in mind is following: user would like to create volume type encryption via openstack4j, and they use this builder method to explicitly set that to our "unspecified" values. Which is currently `0`. I suspect, that we will actually send `key_size: 0` to the openstack hoping it would understand it as `None`, which I am not confident it will do

Reply:
Just a doubt, if users don't want to specify any key size, then they could just skip the key size parameter in builder right? like below:
VolumeTypeEncryptionBuilder.provider("provider") .cipher("cipher").build()
Why would they specify something that they don't want in builder?

Reply:
It is not that they cannot workaround that, I am trying to point out that the code permit us to feed the local entity with data we cannot send. Elsewhere, for fields that can be missing is this value is represented by `null` that gives a hit to a user and it is correctly handled by Jackson, as well. I suspect `0` does none of that.

Reply:
Okay, I will modify int to Integer everywhere for keysize

Reply:
Modified code and verified logic by specifying null for keySize in builder. It worked and I could also retrieve the key size value as null after encryption is created.

Reply:
@olivergondza kindly review the pull request now.